### ISSUE TITLE
[Bug] rule specific simlayer threshold/conditions setting placed at the wrong place

### ISSUE THREAD

**Issue by yqrashawn** [yqrashawn/GokuRakuJoudo#30]
```clojure
{:des   "Vi Mode [D as Trigger Key]"
                       :rules [:vi-mode
                               [:##j :down_arrow nil {:params {:sim 432}}]
                               ....
                               ]}
```
should generates
```json
{
  "description": "Vi Mode [D as Trigger Key]",
  "manipulators": [
    {
      "conditions": [
        {
          "name": "vi-mode",
          "type": "variable_if",
          "value": 1
        }
      ],
      "from": {
        "key_code": "j",
        "modifiers": {
          "optional": [
            "any"
          ]
        }
      },
      "to": [
        {
          "key_code": "down_arrow"
        }
      ],
      "type": "basic"
    },
    {
      "from": {
        "simultaneous": [
          {
            "key_code": "d"
          },
          {
            "key_code": "j"
          }
        ],
        "simultaneous_options": {
          "detect_key_down_uninterruptedly": true,
          "key_down_order": "strict",
          "key_up_order": "strict_inverse",
          "key_up_when": "any",
          "to_after_key_up": [
            {
              "set_variable": {
                "name": "vi-mode",
                "value": 0
              }
            }
          ]
        }
      },
      "parameters": {
        "basic.simultaneous_threshold_milliseconds": 432
      },
      "to": [
        {
          "set_variable": {
            "name": "vi-mode",
            "value": 1
          }
        },
        {
          "key_code": "down_arrow"
        }
      ],
      "type": "basic"
    },
    ......
  ]
}
```
but generates
```json
{
  "description": "Vi Mode [D as Trigger Key]",
  "manipulators": [
    {
      "conditions": [
        {
          "name": "vi-mode",
          "type": "variable_if",
          "value": 1
        }
      ],
      "from": {
        "key_code": "j",
        "modifiers": {
          "optional": [
            "any"
          ]
        }
      },
      "parameters": {
        "basic.simultaneous_threshold_milliseconds": 432
      },
      "to": [
        {
          "key_code": "down_arrow"
        }
      ],
      "type": "basic"
    },
    {
      "from": {
        "simultaneous": [
          {
            "key_code": "d"
          },
          {
            "key_code": "j"
          }
        ],
        "simultaneous_options": {
          "detect_key_down_uninterruptedly": true,
          "key_down_order": "strict",
          "key_up_order": "strict_inverse",
          "key_up_when": "any",
          "to_after_key_up": [
            {
              "set_variable": {
                "name": "vi-mode",
                "value": 0
              }
            }
          ]
        }
      },
      "parameters": {
        "basic.simultaneous_threshold_milliseconds": 250
      },
      "to": [
        {
          "set_variable": {
            "name": "vi-mode",
            "value": 1
          }
        },
        {
          "key_code": "down_arrow"
        }
      ],
      "type": "basic"
    },
    ......
  ]
}
```

should put
```
"parameters": {
        "basic.simultaneous_threshold_milliseconds": 432
},
```
in the both sim rules.

**Comment by materliu** [comment]
why don't flow 
:sim    
config global ? 
use 250

**Comment by materliu** [comment]
ping !

**Comment by yqrashawn** [comment]
Same problem for conditions in simlayer.
```clojure
{
    :applications {:iterm  ["^com\\.googlecode\\.iterm2$"]}
    :simlayers {:o_layer {:key :o}}
    :main [
        {:des "testing" :rules [:o_layer
            [:t :a :iterm]
            [:t :b :!iterm]
        ]}
    ]
}
```
the `"bundle_identifiers" : [ "^com\\.googlecode\\.iterm2$" ]` should be in both config
```json
{
          "type" : "basic",
          "parameters" : {
            "basic.simultaneous_threshold_milliseconds" : 250
          },
          "to" : [ {
            "set_variable" : {
              "name" : "o_layer",
              "value" : 1
            }
          }, {
            "key_code" : "a"
          } ],
          "from" : {
            "simultaneous" : [ {
              "key_code" : "o"
            }, {
              "key_code" : "t"
            } ],
            "simultaneous_options" : {
              "detect_key_down_uninterruptedly" : true,
              "key_down_order" : "strict",
              "key_up_order" : "strict_inverse",
              "key_up_when" : "any",
              "to_after_key_up" : [ {
                "set_variable" : {
                  "name" : "o_layer",
                  "value" : 0
                }
              } ]
            }
          }
        }
```
```json
{
          "from" : {
            "key_code" : "t"
          },
          "to" : [ {
            "key_code" : "a"
          } ],
          "conditions" : [ {
            "name" : "o_layer",
            "value" : 1,
            "type" : "variable_if"
          }, {
            "bundle_identifiers" : [ "^com\\.googlecode\\.iterm2$" ],
            "type" : "frontmost_application_if"
          } ],
          "type" : "basic"
        },
```

**Comment by lrusnac** [comment]
@yqrashawn Can I ask why there are 2 sim rules? 
I somehow expected it to be only the latest one

**Comment by yqrashawn** [comment]
The `simlayer` is a layer that with a simultaneous trigger.
Say we define a `simlayer` named `o_layer` and map `t` to `a` in it.

To karabiner that means two rules.

1. a rule to set `o_layer` to 1 and insert `a` if pressed `ot`.
2. a rule to insert `a` if already in `o_layer` and `t` pressed.

**Comment by seeM** [comment]
With current behaviour, you have to press `ot` twice in a non-iterm application for it to register as `b`, the first time registers as `a`. Is that also due to the bug in [this comment](https://github.com/yqrashawn/GokuRakuJoudo/issues/30#issuecomment-588594233)?

I'm using the `:same_wind_app` idea from your dotfiles, and you have to press the app launch key twice when inside the app to switch between windows.

**Comment by yqrashawn** [comment]
fixed in v0.3.0
